codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2HeadersFrame.java

```diff
@@ -63,9 +63,14 @@ public DefaultHttp2HeadersFrame(Http2Headers headers, boolean endStream, int pad
     }
 
     @Override
-    public DefaultHttp2HeadersFrame setStream(Object stream) {
-      super.setStream(stream);
-      return this;
+    public DefaultHttp2HeadersFrame setStreamId(int streamId) {
```

Comment:
what are thoughts about method signatures that expect returning `this` for non-builder objects? They can be difficult to maintain and often require a method be overridden just to return the correct type. I wonder if it is just simpler to just keep these methods `void` ... it may provide the opportunity for more "fluid" code but sometimes this can also be messy. @ejona86 @buchgr @normanmaurer @nmittler - WDYT?